src/persist-client/src/batch.rs

```diff
@@ -807,48 +819,44 @@ impl<T: Timestamp + Codec64> BatchParts<T> {
                         .inc_by(stats_step_timing.as_secs_f64());
                     stats
                 });
-                (payload_len, stats)
+
+                HollowBatchPart {
+                    key: partial_key,
+                    encoded_size_bytes: payload_len,
+                    key_lower: key_lower.unwrap_or_else(Vec::new),
```

Comment:
feels like key_lower wants to be an Option? i'm a touch uncomfortable using empty as a sentinel for unset here and proto can roundtrip the unset/empty/set distinction on bytes fields (I think you need to add "optional" in the field definition to convince prost to do this), so may as well

Reply:
Normally I loathe a sentinel... but in this case the empty vec is legitimately a valid lower bound for any part, so clients do not need to special-case this value, and leaving it non-optional slightly simplifies usage.

However, if you still prefer making it optional after reading this, that's no problem!

Reply:
Done!